File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

/**
 * @author sx
 */
public interface BrandService {
    /**
     * 查找所有品牌
     *
     * @return 查找结果
     */
    public List<TbBrand> findAll();

    /**
     * 品牌分页
     * @param pageNum 当前页面
     * @param pageSize 每页记录数
     * @return
     */
    public PageResult findPage(int pageNum,int pageSize);

    /**
     *  增加方法
     * @param brand
     */
    public void add(TbBrand brand);

    /**
     * 修改信息
     * @param brand
     */
    public void update(TbBrand brand);

    /**
     * 根据id查找实体单一品牌
     * @param id
     * @return
     */
    public TbBrand findOne(long id);

    /**
     * 多选删除品牌
     * @param ids
     */
    public void delete(long[] ids);

    /**
     * 品牌条件查询
     * @param brand 品牌列表
     * @param pageNum   当前页面
     * @param pageSize 每页记录数
     * @return
     */
    public PageResult findPage(TbBrand brand,int pageNum,int pageSize);

    /**
     *  引入TypeTemplate中的品牌下拉列表
     * @return map
     */
    public List<Map> selectOptionList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//品牌服务
app.service("brandService", function ($http) {
    //查询品牌列表
    this.findAll = function () {
        return $http.get('../brand/findAll.do');
    };
    //分页
    this.findPage = function (page, rows) {
        return $http.get("../brand/findPage.do?page=" + page + "&rows=" + rows);
    }
    //查询单一品牌id
    this.findOne = function (id) {
        return $http.get("../brand/findOne.do?id=" + id);
    }
    //增加
    this.add = function (entity) {
        return $http.post("../brand/add.do", entity);
    }
    //修改
    this.update = function (entity) {
        return $http.post("../brand/update.do", entity);
    }
    //删除
    this.dele=function (ids) {
        return $http.get("../brand/delete.do?ids=" + ids);
    }
    //条件查询
    this.search=function (page,rows,searchEntity) {
        return $http.post("../brand/search.do?page=" + page + "&rows=" + rows, searchEntity);
    }
    //引入TypeTemplate中的品牌下拉列表
    //列表数据
    this.selectOptionList=function () {
        return $http.get("../brand/selectOptionList.do");
    }



});